### ISSUE TITLE
[Bug]: Realms.Exceptions.RealmException: Access to invalidated Collection object

### ISSUE THREAD

**Issue by PawKanarek** [realm/realm-dotnet#2888]
### What happened?

After updating objects in local RealmDB im getting exception `Realms.Exceptions.RealmException: Access to invalidated Collection object `. Sadly the stacktrace of that exception is from Xamarin.Forms/Realm namespaces and i don't know how to prevent my app from crashing. 

Any ideas to workaround this issue will be helpful :) 

### Repro steps

1. Create nested collectionViews in Xamarin
2. Open Realm on main thread 
3. Bind Realm Objects to nested collectionView
4. Update Data from background thread

### Version

Xamarin.iOS 15.6.0.3

### What SDK flavour are you using?

Local Database only

### What type of application is this?

Xamarin

### Client OS and version

iOS

### Code snippets

This exception happens on in my app that's pretty complicated. When i tried to reproduce this issue on empty Xamarin Forms Project, with much simpler data sets then i've got the same exception that was solved in this bug https://github.com/realm/realm-dotnet/issues/2840.

Below snippet will generate `Realms.Exceptions.RealmInvalidObjectException: Can't enumerate the collection because it was deleted or the Realm it is contained in has been closed.`.

Sadly I don't know how to generate `Realms.Exceptions.RealmException: Access to invalidated Collection object` in simple snippet. And im afraid that fix for #2840 won't help for my app & this type of exception. 
```csharp
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Threading.Tasks;
using Realms;
using Xamarin.Forms;

namespace XamarinRealmApp
{
    public class MainPage2 : ContentPage
    {
        public MainPage2()
        {
            // Page Content is CollectionView inside CollectionView
            var collectionView = new CollectionView()
            {
                ItemTemplate = new DataTemplate(() =>
                {
                    var innerCollectionView = new CollectionView()
                    {
                        ItemsLayout = new LinearItemsLayout(ItemsLayoutOrientation.Horizontal),
                        ItemTemplate = new DataTemplate(() =>
                        {
                            var label = new Label { TextColor = Color.Red, WidthRequest = 50};
                            label.SetBinding(Label.TextProperty, nameof(InnerEmbedded.Index));
                            return label;
                        })
                    };
                    // Set binding to inner collection
                    innerCollectionView.SetBinding(ItemsView.ItemsSourceProperty, nameof(OuterEmbedded.InnerItems));
                    return innerCollectionView;
                })
            };
            Content = collectionView;

            // create RealmItem and set it as item source for nested collectionView
            var realm = GetRealm;
            var realmItem = new RealmItem { Id = "Id" };
            realm.Write(() => realm.Add(realmItem, true));
            collectionView.ItemsSource = realmItem.OuterCollection.AsRealmCollection();

            realm.Write(() =>
            {
                var outer = new OuterEmbedded();
                var inner = new InnerEmbedded { Index = 1 };
                outer.InnerItems.Add(inner);
                realmItem.OuterCollection.Add(outer);
            });
            
            // this will crash
            _ = Task.Run(async () =>
            {
                await Task.Delay(1000).ConfigureAwait(false);
                realm = GetRealm;
                realm.Write(() =>
                {
                    realmItem = realm.All<RealmItem>().First();
                    realmItem.OuterCollection.Clear();
                });
            });
        }
        private Realm GetRealm => Realm.GetInstance(new RealmConfiguration(Path.Combine(Environment.CurrentDirectory, "realm.realm")));
    }
    public class RealmItem : RealmObject
    {
        [PrimaryKey] public string Id { get; set; }
        public IList<OuterEmbedded> OuterCollection { get; }
    }
    public class OuterEmbedded : EmbeddedObject
    {
        public IList<InnerEmbedded> InnerItems { get; }
    }
    public class InnerEmbedded : EmbeddedObject
    {
        public int Index { get; set; }
    }
}
```

```csproj
 <ItemGroup>
        <PackageReference Include="Fody" Version="6.6.0">
          <PrivateAssets>all</PrivateAssets>
          <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
        </PackageReference>
        <PackageReference Include="Realm" Version="10.10.0" />
        <PackageReference Include="Realm.Fody" Version="10.10.0">
          <PrivateAssets>all</PrivateAssets>
          <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
        </PackageReference>
        <PackageReference Include="Xamarin.Forms" Version="5.0.0.2012" />
    </ItemGroup>
```

### Stacktrace of the exception/crash you're getting

```shell
Realms.Exceptions.RealmException: Access to invalidated Collection object
  at Realms.NativeException.ThrowIfNecessary (System.Func`2[T,TResult] overrider) [0x0000a] in D:\a\realm-dotnet\realm-dotnet\Realm\Realm\Native\NativeException.cs:55 
  at Realms.CollectionHandleBase.Snapshot () [0x00009] in D:\a\realm-dotnet\realm-dotnet\Realm\Realm\Handles\CollectionHandleBase.cs:39 
  at Realms.RealmCollectionBase`1+Enumerator[T]..ctor (Realms.RealmCollectionBase`1[T] parent) [0x00031] in D:\a\realm-dotnet\realm-dotnet\Realm\Realm\DatabaseTypes\RealmCollectionBase.cs:485 
  at Realms.RealmCollectionBase`1[T].GetEnumerator () [0x00000] in D:\a\realm-dotnet\realm-dotnet\Realm\Realm\DatabaseTypes\RealmCollectionBase.cs:416 
  at Realms.RealmCollectionBase`1[T].System.Collections.IEnumerable.GetEnumerator () [0x00000] in D:\a\realm-dotnet\realm-dotnet\Realm\Realm\DatabaseTypes\RealmCollectionBase.cs:418 
  at Xamarin.Forms.Platform.iOS.ObservableItemsSource.ItemsCount () [0x00018] in D:\a\1\s\Xamarin.Forms.Platform.iOS\CollectionView\ObservableItemsSource.cs:233 
  at Xamarin.Forms.Platform.iOS.ObservableItemsSource.Reload () [0x00007] in D:\a\1\s\Xamarin.Forms.Platform.iOS\CollectionView\ObservableItemsSource.cs:144 
  at Xamarin.Forms.Platform.iOS.ObservableItemsSource.CollectionChanged (System.Collections.Specialized.NotifyCollectionChangedEventArgs args) [0x00067] in D:\a\1\s\Xamarin.Forms.Platform.iOS\CollectionView\ObservableItemsSource.cs:133 
  at Xamarin.Forms.Platform.iOS.ObservableItemsSource.CollectionChanged (System.Object sender, System.Collections.Specialized.NotifyCollectionChangedEventArgs args) [0x0002d] in D:\a\1\s\Xamarin.Forms.Platform.iOS\CollectionView\ObservableItemsSource.cs:108 
  at (wrapper delegate-invoke) <Module>.invoke_void_object_NotifyCollectionChangedEventArgs(object,System.Collections.Specialized.NotifyCollectionChangedEventArgs)
  at Realms.RealmCollectionBase`1[T].RaiseCollectionChanged (System.Collections.Specialized.NotifyCollectionChangedEventArgs args) [0x00000] in D:\a\realm-dotnet\realm-dotnet\Realm\Realm\DatabaseTypes\RealmCollectionBase.cs:343 
  at Realms.RealmCollectionBase`1[T].OnChange (Realms.IRealmCollection`1[T] sender, Realms.ChangeSet change, System.Exception error) [0x00031] in D:\a\realm-dotnet\realm-dotnet\Realm\Realm\DatabaseTypes\RealmCollectionBase.cs:268 
  at Realms.RealmCollectionBase`1[T].Realms.INotifiable<Realms.NotifiableObjectHandleBase.CollectionChangeSet>.NotifyCallbacks (System.Nullable`1[T] changes, System.Nullable`1[T] exception) [0x00162] in D:\a\realm-dotnet\realm-dotnet\Realm\Realm\DatabaseTypes\RealmCollectionBase.cs:412 
  at Realms.NotifiableObjectHandleBase.NotifyObjectChanged (System.IntPtr managedHandle, System.IntPtr changes, System.IntPtr exception) [0x00017] in D:\a\realm-dotnet\realm-dotnet\Realm\Realm\Handles\NotifiableObjectHandleBase.cs:62 
  at (wrapper native-to-managed) Realms.NotifiableObjectHandleBase.NotifyObjectChanged(intptr,intptr,intptr)
  at (wrapper managed-to-native) Realms.SynchronizationContextScheduler.scheduler_invoke_function(intptr,bool)
  at Realms.SynchronizationContextScheduler+Scheduler.<Post>b__4_0 (System.Object f_ptr) [0x00000] in D:\a\realm-dotnet\realm-dotnet\Realm\Realm\Native\SynchronizationContextScheduler.cs:68 
  at Foundation.NSAsyncSynchronizationContextDispatcher.Apply () [0x00000] in /Users/builder/azdo/_work/1/s/xamarin-macios/src/Foundation/NSAction.cs:178 
  at (wrapper managed-to-native) UIKit.UIApplication.UIApplicationMain(int,string[],intptr,intptr)
  at UIKit.UIApplication.Main (System.String[] args, System.String principalClassName, System.String delegateClassName) [0x00013] in /Users/builder/azdo/_work/1/s/xamarin-macios/src/UIKit/UIApplication.cs:73 
  at InsysGo.App.XF.iOS.Application.Main (System.String[] args) [0x0000d] in /Users/pkanarek/Git/insysgo-sdk-mobile/src/Apps/InsysGo/InsysGo.App.XF.iOS/Main.cs:22 
--- End of stack trace from previous location where exception was thrown ---

  at (wrapper managed-to-native) Realms.SynchronizationContextScheduler.scheduler_invoke_function(intptr,bool)
  at Realms.SynchronizationContextScheduler+Scheduler.<Post>b__4_0 (System.Object f_ptr) [0x00000] in D:\a\realm-dotnet\realm-dotnet\Realm\Realm\Native\SynchronizationContextScheduler.cs:68 
  at Foundation.NSAsyncSynchronizationContextDispatcher.Apply () [0x00000] in /Users/builder/azdo/_work/1/s/xamarin-macios/src/Foundation/NSAction.cs:178 
  at (wrapper managed-to-native) UIKit.UIApplication.UIApplicationMain(int,string[],intptr,intptr)
  at UIKit.UIApplication.Main (System.String[] args, System.String principalClassName, System.String delegateClassName) [0x00013] in /Users/builder/azdo/_work/1/s/xamarin-macios/src/UIKit/UIApplication.cs:73 
  at InsysGo.App.XF.iOS.Application.Main (System.String[] args) [0x00037] in /Users/pkanarek/Git/insysgo-sdk-mobile/src/Apps/InsysGo/InsysGo.App.XF.iOS/Main.cs:27
```

**Comment by papafe** [comment]
Hi @PawKanarek, thanks for your report.
We've published a new version (10.11) just yesterday. Would you mind updating your project and letting me know if you still have the same issue?

**Comment by PawKanarek** [comment]
@papafe  With version 10.11 crash dosn't occour! Thank you!
